### PR TITLE
Updates to release instructions

### PR DESCRIPTION
**Pull request by jasongrout**
See also #4377

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -14,6 +14,21 @@ We publish the npm packages, a Python source package, and a Python universal bin
 See the Python docs on [package uploading](https://packaging.python.org/guides/tool-recommendations/)
 for twine setup instructions and for why twine is the recommended method.
 
+## Getting a clean environment
```
Comment (ian-r-rose) [inline RELEASE.md]: I wonder if creating a fully clean environment is overkill. What does this buy that a `git clean -fdx` does not?

Comment (jasongrout) [reply RELEASE.md]: It gets rid of tags. Remember, that caused an abort in my attempt last week, that I had old tags hanging around?

Also, it makes sure that when we later push tags, we don't push any of our possibly personal tags to the public repo.

Comment (ian-r-rose) [reply RELEASE.md]: For the sake of our future selves, can we add a note to that effect, since there are other ways to remove the offending tags?

Comment (jasongrout) [reply RELEASE.md]: Done

Comment (ian-r-rose) [reply RELEASE.md]: :+1:

**Pull request merged**